src/extension.ts

```diff
@@ -25,6 +25,8 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
 
   const lsPath = new ServerPath(context);
   clientHandler = new ClientHandler(lsPath, outputChannel, reporter);
+  clientHandler.extSemanticTokenTypes = tokenTypesFromExtManifest(context.extension);
```

Comment:
We could pass `manifest` (which we declare in line 21) to both functions.

Reply:
I did consider that actually but I felt that we'd loose the type-safety for the argument of both functions as we'd have to make it `any` (at least I couldn't find a way of getting the JSON as a special type) - which makes it more error-prone IMO, but I may be overly sensitive to this. Maybe I'm trying to bring too much Go-ism into TypeScript? 🤷🏻  😄

Reply:
I see your point and think vscode doesn't ship types for the `package.json` content.

We _could_ define a custom interface that only contains the fields we access in the function body, but we would still pass `manifest,` which is `any` to the functions.

```ts
interface PartialManifest {
 contributes: {
   semanticTokenTypes?: ObjectWithId[];
   semanticTokenModifiers?: ObjectWithId[];
 };
}

function tokenTypesFromExtManifest(manifest: PartialManifest): string[] {
``` 

I don't have a clear preference here and would love to have types for the `package.json` content. ([some](https://github.com/sindresorhus/type-fest/blob/main/source/package-json.d.ts) are available, but they lack the vscode specific `contributes` object)

Reply:
Good idea!